### PR TITLE
[19983] Publish Fast-DDS's Logs 

### PR DESCRIPTION
**Pull request by Tempate**
In this version, users can configure through the YAML the publication of Fast DDS's logs. 

Merge after:
- https://github.com/eProsima/dev-utils/pull/92
- https://github.com/eProsima/DDS-Pipe/pull/86
- https://github.com/eProsima/dev-utils/pull/97

### CODE REVIEW

File: ddspipe_core/include/ddspipe_core/configuration/DdsPipeLogConfiguration.hpp [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    virtual bool is_valid(
+            utils::Formatter& error_msg) const noexcept override;
+
+    /////////////////////////
+    // VARIABLES
+    /////////////////////////
+
+    DdsPublishingConfiguration publish;
+
+    bool stdout_enable = false;
```
Comment (LuciaEchevarria99) [inline ddspipe_core/include/ddspipe_core/configuration/DdsPipeLogConfiguration.hpp:54]: If we don't change stdout in YAML but set --log-verbosity info in the command-line, will the logs still not be displayed? This implies that if you wish to configure them via the command-line, you'll also need to set stdout=true in YAML. I notice a slight inconsistency in having to configure them in both places. Perhaps this variable could default to true if the log and/or filter are configured via the command-line?

File: ddspipe_yaml/test/unittest/yaml_reader/log_configuration/YamlReaderLogConfiguration.cpp
Comment (LuciaEchevarria99) [inline ddspipe_yaml/test/unittest/yaml_reader/log_configuration/YamlReaderLogConfiguration.cpp:1]: Add a test with the case:
            R"(
            stdout: false
            verbosity: info
        )"
and check if logs are displayed

**Review by LuciaEchevarria99 — CHANGES_REQUESTED** [review]

File: ddspipe_core/src/cpp/logging/DdsLogConsumer.cpp [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+        throw utils::InitializationException(
+                  utils::Formatter() << "Error creating DataWriter for Participant " <<
+                      participant->guid() << " in topic " << topic << ".");
+    }
+}
+
+void DdsLogConsumer::Consume(
+        const utils::Log::Entry& entry)
+{
+    if (!accept_entry_(entry))
```
[the remark this answers is not in the document]
Comment (Tempate) [reply ddspipe_core/src/cpp/logging/DdsLogConsumer.cpp:108]: https://medium.com/swlh/return-early-pattern-3d18a41bba8

File: ddspipe_core/include/ddspipe_core/logging/DdsLogConsumer.hpp [hunk 3]
```diff
@@ -44,20 +43,16 @@ namespace core {
 /**
  * DDS Log Consumer with Standard (logical) behaviour.
  *
- * This consumer publishes every log entry whose kind is higher or equal to a given one and whose content or category
- * match a regex filter.
- *
- * @attention The consumer filters the entries that it receives, but entries could be filtered beforehand by Fast DDS's
- * Log. To avoid this, set the verbosity to Info and do not apply a filter.
+ * Registering this consumer in Fast DDS's Log publishes the log entries accepted by the BaseLogConsumer.
```
Comment (LuciaEchevarria99) [inline ddspipe_core/include/ddspipe_core/logging/DdsLogConsumer.hpp:46]: Is this sentence completely accurate?

**Review by LuciaEchevarria99 — CHANGES_REQUESTED** [review]

**Review by LuciaEchevarria99 — CHANGES_REQUESTED** [review]

File: ddspipe_core/include/ddspipe_core/logging/DdsLogConsumer.hpp (hunk 3 above)
Comment (Tempate) [reply ddspipe_core/include/ddspipe_core/logging/DdsLogConsumer.hpp:46]: Yes, why do you think it isn't?

File: ddspipe_core/CMakeLists.txt
[the remark this answers is not in the document]
Comment (Tempate) [reply ddspipe_core/CMakeLists.txt:1]: I will rebase once it's fully approved and merged.

**Pull request merged**